src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/GitFileManager.cs

```diff
@@ -806,5 +885,96 @@ void BuildDependencies(XmlNodeList dependencies)
 
             return dependencyDetails.Where(d => !d.Pinned);
         }
+
+        private HashSet<string> FlattenLocations(Dictionary<string, HashSet<string>> assetLocationMap, Func<string, bool> filter = null)
+        {
+            HashSet<string> managedFeeds = new HashSet<string>();
+
+            foreach (HashSet<string> locations in assetLocationMap.Values)
+            {
+                IEnumerable<string> filteredLocations = filter != null ?
+                    locations.Where(filter) :
+                    locations;
+
+                managedFeeds.UnionWith(filteredLocations);
+            }
+            return managedFeeds;
+        }
+
+        private List<(string key, string feed)> GetPackageSources(XmlDocument nugetConfig, Func<string, bool> filter = null)
+        {
+            var sources = new List<(string key, string feed)>();
+            XmlNodeList nodes = nugetConfig.SelectNodes("//configuration/packageSources/add");
+            foreach (XmlNode node in nodes)
+            {
+                if (node.NodeType != XmlNodeType.Comment && node.NodeType != XmlNodeType.Whitespace)
+                {
+                    var keyContent = node.Attributes["key"]?.Value;
+                    var valueContent = node.Attributes["value"]?.Value;
+                    if (keyContent != null && valueContent != null)
+                    {
+                        if (filter != null && !filter(valueContent))
+                        {
+                            continue;
+                        }
+                        sources.Add((keyContent, valueContent));
+                    }
+                }
+            }
+            return sources;
+        }
+
+        private List<(string key, string feed)> GetManagedPackageSources(HashSet<string> feeds)
+        {
+            var sources = new List<(string key, string feed)>();
+
+            foreach (string feed in feeds)
+            {
+                var parsedFeed = ParseManagedPackageFeed(feed);
+
+                string key = $"darc-{parsedFeed.repoName}-{parsedFeed.sha.Substring(0, 7)}";
```

Comment:
>darc- [](start = 31, length = 5)

"int|pub-" is missing here

Reply:
You think that's important to have in the key?

Reply:
discussed offline. Might as well add it. It's a cheap change